### PR TITLE
import plugin refactor

### PR DESCRIPTION
**Pull request by etokareva**

### CODE REVIEW

**Review by vkorolev — APPROVED** [review]

File: plugin/import/import.view.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import {EventListener, EventManager} from '@grid/core/infrastructure';
+import {upload} from '@grid/core/services/upload';
+import {readFile} from './read';
+
+export class ImportView extends PluginView {
+	constructor(model, context) {
+		super(...arguments);
+		this.importOptions = context.importOptions;
+		const element = context.element;
+		this.eventListener = new EventListener(element, new EventManager(this));
```
Comment (klumba12) [inline plugin/import/import.view.js]: make this outside of *.view different frameworks can work differently

File: plugin/import/import.view.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import {PluginView} from '../plugin.view';
+import {Command} from '@grid/core/command';
+import {EventListener, EventManager} from '@grid/core/infrastructure';
+import {upload} from '@grid/core/services/upload';
+import {readFile} from './read';
+
+export class ImportView extends PluginView {
+	constructor(model, context) {
+		super(...arguments);
+		this.importOptions = context.importOptions;
```
Comment (klumba12) [inline plugin/import/import.view.js]: newline + rename import options to options

File: plugin/import/import.view.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		for (let file of files) {
+			const reader = new FileReader();
+			reader.onload = e => {
+				readFile(e, file, this.model, this.importOptions);
+			};
+			reader.readAsBinaryString(file);
+		}
+	}
+
+	get resource() {
```
Comment (klumba12) [inline plugin/import/import.view.js]: move it to plugin itself

File: plugin/import/import.view.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import {Command} from '@grid/core/command';
+import {upload} from '@grid/core/services/upload';
+import {readFile} from './read';
+
+export class ImportView extends PluginView {
+	constructor(model, context) {
+		super(...arguments);
+
+		const element = context.element;
+		const eventListener = context.eventListener;
```
Comment (klumba12) [inline plugin/import/import.view.js]: Kate can you put event call functionality to plugin itself? 
`.on(() =>$import.load()...`

File: src/plugins/import/import.js [hunk 5]
```diff
@@ -15,41 +13,30 @@ TemplatePath
...
-		this.using(this.eventListener.on('change', this.load));
-	}
 }
 
 export default Import.component({
 	controller: Import,
-	controllerAs: '$import',
+	controllerAs: '$importPlugin',
 	bindings: {
 		'importOptions': '='
```
Comment (klumba12) [inline src/plugins/import/import.js]: rename it to options too

**Review by klumba12 — APPROVED** [review]

**Pull request merged**